lib/ota/zigbeeOTA.js

```diff
@@ -11,13 +11,15 @@ async function getImageMeta(current, logger, device) {
     const modelId = device.modelID;
     const imageType = current.imageType;
     const manufacturerCode = current.manufacturerCode;
+    const manufacturerName = device.manufacturerName;
     const images = (await axios.get(url)).data;
 
     // NOTE: Officially an image can be determined with a combination of manufacturerCode and imageType.
     // However Gledopto pro products use the same imageType (0) for every device while the image is different.
     // For this case additional identification through the modelId is done.
+    // In the case of Tuya end Moes, additional identification is carried out through the manufacturerName.
     const image = images.find((i) => i.imageType === imageType && i.manufacturerCode === manufacturerCode &&
-        (!i.modelId || i.modelId === modelId));
+        (!i.modelId || i.modelId === modelId) && (!i.manufacturerName || i.manufacturerName === manufacturerName));
```

Comment:
This line should be changed to

```js
(!i.modelId || i.modelId === modelId) && (!i.manufacturerName || i.manufacturerName.includes(manufacturerName)));
```

After that, can you check if everything works, then I can merge it.